scipy/stats/_multivariate.py

```diff
@@ -385,11 +388,32 @@ def __call__(self, mean=None, cov=1, allow_singular=False, seed=None):
                                           allow_singular=allow_singular,
                                           seed=seed)
 
-    def _process_parameters(self, dim, mean, cov):
+    def _process_parameters(self, mean, cov, allow_singular=True):
         """
         Infer dimensionality from mean or covariance matrix, ensure that
         mean and covariance are full vector resp. matrix.
         """
+        if isinstance(cov, _covariance.Covariance):
+            dim, mean, cov_object = self._process_parameters_new(mean, cov)
+            return dim, mean, cov_object
+        else:
+            dim, mean, cov = self._process_parameters_old(None, mean, cov)
+            psd = _PSD(cov, allow_singular=allow_singular)
+            cov_object = _covariance.CovViaPSD(psd)
+            return dim, mean, cov_object
+
+    def _process_parameters_new(self, mean, cov):
+        dim = cov.dimensionality
+        mean = np.array([0.]) if mean is None else mean
+        message = (f"`cov` represents a covariance matrix in {dim} dimensions,"
+                   f"and so `mean` must be broadcastable to shape {(dim,)}")
+        try:
+            mean = np.broadcast_to(mean, dim)
+        except ValueError as e:
+            raise ValueError(message) from e
+        return dim, mean, cov
+
+    def _process_parameters_old(self, dim, mean, cov):
```

Comment:
It could have been done that way. On one hand, it's one less method. On the other, it modifies an old method. Since it's already written one way, and since there's no clear winner, I'd prefer to leave it alone. I have renamed the methods though.